pelita/game.py

```diff
@@ -837,27 +837,22 @@ def apply_move(gamestate, bot_position):
     # unless we have already made an error, check if we made a legal move
     if not (n_round, turn) in team_errors:
         if bot_position not in legal_positions:
-            error_dict = {
-                "reason": 'illegal move',
-                "bot_position": bot_position
-                }
-            # add the error to the team’s errors
-            game_print(turn, f"Illegal position. {bot_position} not in legal positions: {sorted(legal_positions)}.")
-            team_errors[(n_round, turn)] = error_dict
+            previous_position = gamestate["bots"][gamestate["turn"]]
+            game_print(turn, f"Illegal position. {previous_position}➔{bot_position} not in legal positions:"
+                             f" {sorted(legal_positions)}.")
+            exception_event = {
+                    'type': 'IllegalPosition',
+                    'description': f"bot{turn}: {previous_position}➔{bot_position}",
+                    'turn': turn,
+                    'round': n_round,
+            }
+            gamestate['fatal_errors'][team].append(exception_event)
 
     # only execute move if errors not exceeded
     gamestate.update(check_gameover(gamestate))
     if gamestate['gameover']:
         return gamestate
 
-    # Now check if we must make a random move
```

Comment:
You need this section to handle slow bots. Without it the whole game crashes on a timeout.

Reply:
yep, fixed

Reply:
I forced pushed to recover the code that handled random moves in the face of non-fatal errors. I also re-wrote the corresponding test to use timeouts instead of illegal moves to test that the error limit still works